### PR TITLE
Switch to pytest and include Python 3.10

### PR DESCRIPTION
**Pull request by pgajdos**
I am not completely sure about tox.ini etc.

### CODE REVIEW

**Comment by spacegaier** [comment]
What would the benefit of that switch? I haven't looked myself into the differences between `nosetest` and `pytest`, but I assume you have. Could you provide your reasoning for advocating that change in `voluptuous`?

**Comment by pgajdos** [comment]
nose is deprecated
https://nose.readthedocs.io/en/latest/index.html

File: voluptuous/tests/tests.py [hunk 1]
```diff
@@ -31,18 +30,16 @@ def test_exact_sequence():
...
     schema = Schema({Required('q'): 1})
-    # Can't use nose's raises (because we need to access the raised
-    # exception, nor assert_raises which fails with Python 2.6.9.
     try:
         schema({})
     except Invalid as e:
-        assert_equal(str(e), "required key not provided @ data['q']")
+        assert str(e) == "required key not provided @ data['q']"
     else:
         assert False, "Did not raise Invalid"
```
Comment (freundTech) [inline voluptuous/tests/tests.py:44]: You could use `pytest.raises` here and for all other exceptions:

```
with pytest.raises(Invalid) as e:
    schema({})
assert str(e.value) == "required key not provided @ data['q']"

**Comment by spacegaier** [comment]
Superseded by https://github.com/alecthomas/voluptuous/pull/464

**Pull request closed without merging**